catalog/views.py

```diff
@@ -1,3 +1,26 @@
 from django.shortcuts import render
+from .models import Book, Author, BookInstance, Genre
 
-# Create your views here.
+
+def index(request):
+    """View function for home page of site."""
+
+    # Generate counts of some of the main objects
+    num_books = Book.objects.all().count()
+    num_instances = BookInstance.objects.all().count()
+
+    # Available books (status = 'a')
+    num_instances_available = BookInstance.objects.filter(status__exact='a').count()
```

Comment:
Tìm hiểu các Field lookup khác ngoài `exact`

Reply:
các Field Lookups phổ biến trong Django:
exact: Giá trị chính xác
iexact: Giá trị chính xác (case-insensitive)
contains: Giá trị chứa chuỗi
icontains: Giá trị chứa chuỗi (case-insensitive)
gt: Lớn hơn
lt: Nhỏ hơn
gte: Lớn hơn hoặc bằng
lte: Nhỏ hơn hoặc bằng
in: Giá trị nằm trong danh sách
startswith: Bắt đầu bằng
istartswith: Bắt đầu bằng (case-insensitive)
endswith: Kết thúc bằng
iendswith: Kết thúc bằng (case-insensitive)
range: Trong khoảng
year: Truy vấn dựa trên năm (ngày/tháng/năm)
month: Truy vấn dựa trên tháng (ngày/tháng/Năm)
day: Truy vấn dựa trên ngày (ngày/tháng/năm)